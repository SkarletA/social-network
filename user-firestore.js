/* eslint import/no-unresolved: */
import {
  getFirestore,
  doc,
  getDoc,
  setDoc,
  updateDoc,
} from './firebase.js';

import initApp from './initApp.js';

initApp();
const db = getFirestore();

export async function createUser(id, data) {
  const docData = {
    ...data,
  };
  await setDoc(doc(db, 'users', id), docData);
}

export async function getUser(id) {
  return getDoc(doc(db, 'users', id));
}
export async function updateUser(id, newField) {
  return updateDoc(doc(db, 'users', id), newField);
}
