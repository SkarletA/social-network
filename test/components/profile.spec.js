/**
 * @jest-environment jsdom
*/
// import profile from '../../src/views/profile.js';
import { getUser } from '../../src/user-firestore.js';

jest.mock('../../src/firebase.js');

describe('profile', () => {
  const id = 'zysf456';
  it('Deberia retornar el objeto autenticado', async () => {
    const profiles = await getUser(id);
    expect(profiles).toEqual({});
  });

  it('Debería envíar un error al retornar un objeto', async () => {
    try {
      await getUser(id);
    } catch (error) {
      expect(error).toMatch('error');
    }
  });
});
