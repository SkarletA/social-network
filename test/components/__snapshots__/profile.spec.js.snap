// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile Deberia renderizar profile 1`] = `"<div class=\\"container-profile\\"><navbar class=\\"nav-bar\\"><div class=\\"nav-bar-container\\"><ul class=\\"icon-home\\"><li><a data-path=\\"/home\\" id=\\"home\\"><img class=\\"mapask\\" src=\\"https://svgshare.com/i/f6U.svg\\" title=\\"Home\\"></a></li></ul><ul class=\\"icons-bar\\"><li><a id=\\"logOut\\" data-path=\\"/\\"><img src=\\"https://svgshare.com/i/fAy.svg\\" title=\\"LogOut\\" class=\\"log-out\\"></a></li><li><a data-path=\\"/home\\" id=\\"home\\"><img class=\\"home\\" src=\\"https://svgshare.com/i/fEX.svg\\" title=\\"Home\\"></a></li><li><a data-path=\\"/profile\\" id=\\"profileBtn\\"><img class=\\"img-profile\\" src=\\"https://svgshare.com/i/f5r.svg\\"></a></li></ul></div></navbar><section class=\\"container-sections\\"><aside class=\\"hash-second\\"></aside><section class=\\"post-user\\"><form class=\\"form-home\\" id=\\"formHome\\"><textarea class=\\"description-Posts\\" id=\\"descriptionPosts\\" name=\\"description-posts\\" placeholder=\\"Â¿Sobre que quieres hablar?\\"></textarea><button type=\\"submit\\" class=\\"btn-post\\" id=\\"btnPost\\"></button></form><div></div><section class=\\"post-container\\" id=\\"postContainer\\"></section></section><aside class=\\"profile-primary\\"><section class=\\"container-profile-users\\" id=\\"containerProfileUsers\\"><button class=\\"btn-update button\\" id=\\"btnUpdate\\"></button></section></aside></section></div>"`;
